pattern/variable/VariableRegistry.java

```diff
@@ -94,36 +97,33 @@ public static VariableRegistry createFromVariables(List<? extends BoundVariable>
             List<ConceptVariable> unboundedVariables = new ArrayList<>();
             VariableRegistry registry = new VariableRegistry(bounds, allowDerived);
             variables.forEach(typeQLVar -> {
-                if (typeQLVar.isConcept()) unboundedVariables.add(typeQLVar.asConcept());
-                else registry.register(typeQLVar);
+                if (typeQLVar.isThing()) registry.register(typeQLVar.asThing());
+                else if (typeQLVar.isType()) registry.register(typeQLVar.asType());
+                else if (typeQLVar.isValue()) registry.register(typeQLVar.asValue());
+                else if (typeQLVar.isConcept()) unboundedVariables.add(typeQLVar.asConcept());
+                else throw TypeDBException.of(ILLEGAL_STATE);
             });
-            unboundedVariables.forEach(registry::register);
+            unboundedVariables.forEach(registry::register); // Also need to bring back the Base type Variable register method.
```

Comment:
Has this been done?

Reply:
Nope. And I don't think I need to anymore.